File: 001 Array and Hashing/easy/containsDuplicate.py
# LINK : 

# MY WAY 

class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        return len(set(nums)) != len(nums)


# A Better Way 

class Solution:
    def containsDuplicate(self, nums):
        mySet = set()
        for i in nums:
            if i in mySet:
                return True
            mySet.add(i)
        return False

File: 001 Array and Hashing/easy/rust/contains_duplicate.rs
// LINK : https://leetcode.com/problems/contains-duplicate/
use std::collections::HashSet;

impl Solution {
    pub fn contains_duplicate(nums: Vec<i32>) -> bool {
        let mut u_nums = HashSet::new();

        for i in nums {
            if u_nums.contains(&i) {
                return true
            }
            u_nums.insert(i);
        }
        return false
    }
}
